### PR TITLE
Back extensions section alias import

### PR DESCRIPTION
**Pull request by rbatistadev**
#### Type Of Change 
- [ ] Small bug fix
- [X] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [X] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [X] Commits are split per component (schema, web/admin, kamusers, agis, ..)
- [X] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 
- [ ] Upport from existing Pull request #XXXX

### CODE REVIEW

File: web/rest/client/config/routes.yaml [hunk 1]
```diff
@@ -163,6 +163,16 @@ get_hunt_group_users_available:
     _api_collection_operation_name: 'get_hunt_group_users_available'
     _api_receive: false
 
+post_extensions_mass_import:
+  path: '/extensions/mass_import'
```
Comment (mmadariaga) [inline web/rest/client/config/routes.yaml:167]: missing tests

**Review by mmadariaga — CHANGES_REQUESTED** [review]

File: library/Ivoz/Provider/Domain/Service/Extension/ExtensionFactory.php [hunk 2]
```diff
@@ -42,6 +44,19 @@ public function fromMassProvisioningCsv(
                 ExtensionInterface::ROUTETYPE_USER
             );
 
+        if ($countryPrefix) {
+            $extensionDto
+                ->setNumberCountryId(
+                    substr($countryPrefix, 1)
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Service/Extension/ExtensionFactory.php:50]: NumberCountry is a foreign key to Country table. You'll have to map country prefix to it's pk

File: library/phpstan-baseline.neon [hunk 3]
```diff
@@ -1870,6 +1870,16 @@ parameters:
 			count: 1
 			path: Ivoz/Kam/Infrastructure/Persistence/Doctrine/UsersLocationDoctrineRepository.php
 
+		-
+			message: "#^Parameter \\#1 \\$haystack of function str_contains expects string, array\\|string given\\.$#"
```
Comment (mmadariaga) [inline library/phpstan-baseline.neon:1874]: You can fix this

File: web/rest/client/features/provider/extension/postExtension.feature [hunk 4]
```diff
@@ -99,3 +99,35 @@ Feature: Create extensions
...
+    And the response should be in JSON
+    And the header "Content-Type" should be equal to "application/json; charset=utf-8"
+    And the JSON should be equal to:
+    """
+      {
+          "success": true,
+          "errorMsg": "",
+          "failed": 0
+      }
+    """
```
Comment (mmadariaga) [inline web/rest/client/features/provider/extension/postExtension.feature:133]: Retrieve created extensions to ensure everything is fine

File: web/rest/client/src/Controller/Provider/PostExtensionsMassImportAction.php [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        }
+
+        /** @var ?AdministratorInterface $company */
+        $company = $token->getUser();
+
+        if (!$company) {
+            throw new ResourceClassNotFoundException('Company not found');
+        }
+
+        if (!$company->getBrand()) {
```
Comment (mmadariaga) [inline web/rest/client/src/Controller/Provider/PostExtensionsMassImportAction.php:42]: Company has always a brand

File: web/rest/client/src/Controller/Provider/PostExtensionsMassImportAction.php [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+            throw new \DomainException(
+                'Forbidden company',
+                403
+            );
+        }
+
+        $calendarId = $request->request->get('calendar');
```
Comment (mmadariaga) [inline web/rest/client/src/Controller/Provider/PostExtensionsMassImportAction.php:49]: calendar?

File: web/rest/client/src/Model/ExtensionsMassImport.php [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    /**
+     * @var integer
+     * @AttributeDefinition(type="int")
+     */
+    protected $failed;
+
+    public function __construct(
+        bool $success,
+        string $errorMsg,
+        int $imported
```
Comment (mmadariaga) [inline web/rest/client/src/Model/ExtensionsMassImport.php:33]:
```suggestion
        int $failed
```

**Review by mmadariaga — CHANGES_REQUESTED** [review]

File: library/Ivoz/Provider/Domain/Service/Extension/ExtensionFactory.php [hunk 8]
```diff
@@ -42,6 +49,26 @@ public function fromMassProvisioningCsv(
                 ExtensionInterface::ROUTETYPE_USER
             );
 
+        if ($countryCode) {
+            $country = $this->countryRepository
+                ->findOneByCountryCode($countryCode);
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Service/Extension/ExtensionFactory.php:54]: You'll need to use both, CountryPrefix and CountryCode (This one is optional is the csv) ir order to resolve the country.

CSV example:
```
Extension,CountryPrefix,Number,CountryCode
200,+34,944048184,ES
201,+34,944048185,ES
202,+34,944048186
203,+1,944048187
204,+1,944048188,US
```

Countries in db:
```
mysql> select id, code, name_en, countryCode from Countries where countryCode = '+1';
+-----+------+--------------------------------------+-------------+
| id  | code | name_en                              | countryCode |
+-----+------+--------------------------------------+-------------+
|  38 | CA   | Canada                               | +1          |
| 184 | PR   | Puerto Rico                          | +1          |
| 234 | UM   | United States Minor Outlying Islands | +1          |
| 235 | US   | United States of America             | +1          |
+-----+------+--------------------------------------+-------------+
4 rows in set (0,00 sec)
```

**Review by mmadariaga — CHANGES_REQUESTED** [review]

File: web/rest/client/features/provider/extension/postExtensionMassImport.feature [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+              "numberValue": null,
+              "friendValue": null,
+              "id": 4,
+              "ivr": null,
+              "huntGroup": null,
+              "conferenceRoom": null,
+              "user": null,
+              "queue": null,
+              "conditionalRoute": null,
+              "numberCountry": null,
```
Comment (mmadariaga) [inline web/rest/client/features/provider/extension/postExtensionMassImport.feature:180]: should this throw an error?

Comment (rbatistadev) [reply web/rest/client/features/provider/extension/postExtensionMassImport.feature:180]: No it should not.

File: library/Ivoz/Provider/Application/Service/Extension/SyncFromCsv.php [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+                $number = $fields[2];
+                $code = $fields[3] ?? null;
+
+                $extension = $this->extensionFactory->fromMassProvisioningCsv(
+                    $companyId,
+                    (string) $extensionNumber,
+                    countryCode: $countryCode,
+                    number: $number,
+                    code: $code,
+                    user: null
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Application/Service/Extension/SyncFromCsv.php:55]: don't set user as it's always null

File: library/Ivoz/Provider/Application/Service/User/SyncFromCsv.php [hunk 11]
```diff
@@ -85,7 +85,10 @@ public function execute(CompanyInterface $company, string $csv)
                     $extension = $this->extensionFactory->fromMassProvisioningCsv(
                         (int) $company->getId(),
                         $extensionArg,
-                        $user
+                        $user,
+                        null,
+                        null,
+                        null
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Application/Service/User/SyncFromCsv.php:91]: Don't set null values. they are not required

File: library/spec/Ivoz/Provider/Application/Service/User/SyncFromCsvSpec.php [hunk 12]
```diff
@@ -143,7 +143,10 @@ function it_calls_extension_factory()
             ->fromMassProvisioningCsv(
                 $this->company->getId(),
                 '2002',
-                Argument::type(UserInterface::class)
+                Argument::type(UserInterface::class),
+                null,
+                null,
+                null
             )
```
Comment (mmadariaga) [inline library/spec/Ivoz/Provider/Application/Service/User/SyncFromCsvSpec.php:150]: unnecessary

File: web/rest/client/config/api/resources.yml [hunk 13]
```diff
@@ -151,3 +151,25 @@ Model\HolidaysMassImport:
...
+        produces:
+          - 'application/json'
+        tags:
+          - Provider
+        parameters:
+          - name: csv
+            in: formData
+            type: file
+            required: true
+            description: "*extension, *country prefix, number, country code"
```
Comment (mmadariaga) [inline web/rest/client/config/api/resources.yml:174]:
```suggestion
            description: "*extension, *country prefix, *number, country code"
```

**Review by mmadariaga — CHANGES_REQUESTED** [review]

File: library/Ivoz/Provider/Application/Service/Extension/SyncFromCsv.php [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+                $extensionNumber = $fields[0];
+                $countryCode = $fields[1];
+                $number = $fields[2];
+                $code = $fields[3] ?? null;
+
+                $extension = $this->extensionFactory->fromMassProvisioningCsv(
+                    $companyId,
+                    (string) $extensionNumber,
+                    countryCode: $countryCode,
+                    number: $number,
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Application/Service/Extension/SyncFromCsv.php:53]: Dont' mix named arguments and classic arguments

**Review by mmadariaga — CHANGES_REQUESTED** [review]

**Review by mmadariaga — APPROVED** [review]

**Pull request merged**